security_groups.tf

```diff
@@ -0,0 +1,88 @@
+#Security group for Blue server
+resource "aws_security_group" "server_sg" {
+  name        = "server-sg"
+  description = "Allow connection for server security group."
+  vpc_id      = data.aws_vpc.vpc_group3.id
+
+  ingress {
+    description      = "Allow port 22"
+    from_port        = 22
+    to_port          = 22
+    protocol         = "tcp"
+    cidr_blocks      = ["110.224.205.97/32"]
```

Comment:
This is the security group of the ASG instances, so SSH access needs to be allowed only from the bastion host from the VPC/subnet cidr or the bastion's SG id.